src/inmanta/db/versions/v202503030.py

```diff
@@ -0,0 +1,288 @@
+            inmanta_module_name varchar NOT NULL,
+            inmanta_module_version varchar NOT NULL,
+            environment uuid NOT NULL,
+            file_content_hash varchar NOT NULL REFERENCES file(content_hash) ON DELETE CASCADE,
+            python_module_name varchar NOT NULL,
+            is_byte_code boolean NOT NULL,
+            PRIMARY KEY(environment, inmanta_module_name, inmanta_module_version, python_module_name),
+            FOREIGN KEY (environment, inmanta_module_name, inmanta_module_version)
+                REFERENCES public.inmanta_module(environment, name, version) ON DELETE CASCADE
+        );
+
+        CREATE INDEX files_in_module_file_content_hash_index
+        ON public.files_in_module (file_content_hash);
+
+        CREATE TABLE public.modules_for_agent (
+            cm_version integer NOT NULL,
+            agent_name varchar NOT NULL,
+            inmanta_module_name varchar NOT NULL,
+            inmanta_module_version varchar NOT NULL,
+            environment uuid NOT NULL,
+            PRIMARY KEY(environment, cm_version, agent_name, inmanta_module_name, inmanta_module_version),
+            FOREIGN KEY (environment, cm_version)
+                REFERENCES public.configurationmodel(environment, version) ON DELETE CASCADE,
+            FOREIGN KEY (environment, agent_name)
+                REFERENCES public.agent(environment, name) ON DELETE CASCADE,
+            FOREIGN KEY (environment, inmanta_module_name, inmanta_module_version)
+                REFERENCES public.inmanta_module(environment, name, version) ON DELETE CASCADE
+        );
+
+        CREATE INDEX modules_for_agent_environment_agent_name_index
+        ON public.modules_for_agent (environment, agent_name);
+        CREATE INDEX modules_for_agent_environment_module_name_module_version_index
+        ON public.modules_for_agent (environment, inmanta_module_name, inmanta_module_version);
+
+    """
+
+    await connection.execute(schema)
+
+    SourceInfo = namedtuple("SourceInfo", ["file_hash", "file_path", "python_module_name", "requirements"])
+
+    @dataclass
+    class SetOfSources:
+        sources: set[SourceInfo] = field(default_factory=lambda: set())
+
+    @dataclass
+    class SourcesPerModule:
+        # Maps inmanta module names to their sources
+        inmanta_modules: defaultdict[str, SetOfSources] = field(default_factory=lambda: defaultdict(SetOfSources))
+
+    @dataclass
+    class ModulesPerVersion:
+        # Maps model versions to their inmanta modules
+        model_versions: defaultdict[int, SourcesPerModule] = field(default_factory=lambda: defaultdict(SourcesPerModule))
+
+    @dataclass
+    class VersionsPerEnv:
+        # Maps environments to model versions
+        environments: defaultdict[str, ModulesPerVersion] = field(default_factory=lambda: defaultdict(ModulesPerVersion))
+
+    async def fetch_code_data(
+        code_data: VersionsPerEnv,
+        resource_type_to_module: dict[int, dict[str, set[str]]],
```

Comment:
if these are output, I would make them return values, this is very hard to read.

the applies to all code in this file. I can't make out how it works, because input and output are not clearly separated.